src/structlog/tracebacks.py

```diff
@@ -99,20 +107,58 @@ def safe_str(_object: Any) -> str:
         return f"<str-error {str(error)!r}>"
 
 
-def to_repr(obj: Any, max_string: int | None = None) -> str:
-    """Get repr string for an object, but catch errors."""
-    if isinstance(obj, str):
-        obj_repr = obj
+def to_repr(
+    obj: Any, max_length: int | None = None, max_string: int | None = None
+) -> str:
+    """
+    Get repr string for an object, but catch errors.
+
+    :func:`repr()` is used for strings, too, so that secret wrappers that
+    inherit from :func:`str` and overwrite ``__repr__()`` are handled correctly
+    (i.e. secrets are not logged in plain text).
+
+    Args:
+        obj: Object to get a string representation for.
+
+        max_length: Maximum length of containers before abbreviating, or
+            ``None`` for no abbreviation.
+
+        max_string: Maximum length of string before truncating, or ``None`` to
+            disable truncating.
+
+    Returns:
+        The string representation of *obj*.
+
+    .. versionchanged:: 24.3.0
+       Added *max_length* argument.  Use :program:`rich` to render locals if it
+       is available.  Call :func:`repr()` on strings in fallback
+       implementation.
+    """
+    if rich is not None:
+        # Let rich render the repr if it is available.
+        # It produces much better results for containers and dataclasses/attrs.
+        obj_repr = rich.pretty.traverse(
+            obj, max_length=max_length, max_string=max_string
```

Comment:
Wouldn't it be better to make this optional?

Reply:
I can make it optional but using rich should be the default.
If these are the locals: `{"spam": True, "eggs": "Lorem ipsum dolor", "bacon": [1, 2, 3, 4, 5, 5, 7]}`,
then rich would compute:
```
{'spam': True, 'eggs': 'Lorem ipsu'+7, 'bacon': [1, 2, 3, 4, ... +3]}
```
while my simple algorithm would compute:
```
{'spam': True, 'eggs': 'Lorem ipsu'+7, 'bacon': '[1, 2, 3, '+11}
```

The rich algorithm works a lot better with nested structure then my simple variant.

Reply:
I’m 100% with it being the default. I just know people will ask for an off switch. :)

Reply:
Done